smarts/core/sumo_road_network.py

```diff
@@ -409,10 +413,11 @@ def _waypoint_paths_at(
             )
 
         @lru_cache(maxsize=4)
-        def shape(self, buffer_width: float = 0.0) -> Polygon:
-            assert buffer_width >= 0.0
-            if buffer_width > 0:
-                return buffered_shape(self._sumo_lane.getShape(), buffer_width)
+        def shape(self, width: float = 0.0, buffer_width: float = 0.0) -> Polygon:
+            new_width = width + buffer_width
+            assert new_width >= 0.0
+            if new_width > 0:
+                return buffered_shape(self._sumo_lane.getShape(), new_width)
             line = self._sumo_lane.getShape()
             bline = buffered_shape(line, 0.0)
             return line if bline.is_empty else bline
```

Comment:
I have removed the unnecessary branch but the two methods are not duplicate. One is for lane shape and other is for road shape. Both need to defined separately since that's how their parents `RoadMap.Lane` and `RoadMap.Road` are defined.